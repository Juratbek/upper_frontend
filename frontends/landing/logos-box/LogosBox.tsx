import { useDevice } from 'hooks';
import Image from 'next/image';
import { FC, useMemo } from 'react';

import classes from './LogosBox.module.scss';

export const LogosBox: FC = () => {
  const { isDesktop } = useDevice();
  const paths = useMemo(() => {
    if (isDesktop)
      return ['vs_code.png', 'flutter.png', 'adobe.png', 'figma.png', 'chat-gpt.png', 'js.png'];
    return ['vs_code.png', 'chat-gpt.png', 'js.png', 'adobe.png'];
  }, []);

  return (
    <div className={classes.container}>
      {paths.map((path, index) => (
        <Image
          className={classes.image}
          key={path}
          width={120}
          height={120}
          src={`/logos/${path}`}
          alt='vs code'
          // @ts-ignore
          style={{ '--i': index }}
        />
      ))}
    </div>
  );
};
