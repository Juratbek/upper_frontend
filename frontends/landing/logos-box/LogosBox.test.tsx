import { render } from '@testing-library/react';
import { ReduxWrapper } from 'tests';
import { vi } from 'vitest';

import { LogosBox } from './LogosBox';

const mocks = vi.hoisted(() => ({ useDevice: vi.fn().mockReturnValue({ isDesktop: true }) }));

vi.mock('hooks', () => ({
  useDevice: mocks.useDevice,
}));

describe('LogosBox', () => {
  it('snapshot', () => {
    render(
      <ReduxWrapper>
        <LogosBox />
      </ReduxWrapper>,
    );

    expect(document.body).toMatchSnapshot();
  });

  it('snapshot for non-desktop devices', () => {
    mocks.useDevice.mockReturnValue({ isDesktop: false });

    render(
      <ReduxWrapper>
        <LogosBox />
      </ReduxWrapper>,
    );

    expect(document.body).toMatchSnapshot();
  });
});
