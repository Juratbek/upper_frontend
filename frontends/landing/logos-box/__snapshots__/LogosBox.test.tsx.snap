// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogosBox > snapshot 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fvs_code.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75 2x"
        style="color: transparent; --i: 0;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fflutter.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fflutter.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fflutter.png&w=256&q=75 2x"
        style="color: transparent; --i: 1;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fadobe.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75 2x"
        style="color: transparent; --i: 2;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Ffigma.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Ffigma.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Ffigma.png&w=256&q=75 2x"
        style="color: transparent; --i: 3;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75 2x"
        style="color: transparent; --i: 4;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fjs.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fjs.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fjs.png&w=256&q=75 2x"
        style="color: transparent; --i: 5;"
        width="120"
      />
    </div>
  </div>
</body>
`;

exports[`LogosBox > snapshot for non-desktop devices 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fvs_code.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fvs_code.png&w=256&q=75 2x"
        style="color: transparent; --i: 0;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fchat-gpt.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fchat-gpt.png&w=256&q=75 2x"
        style="color: transparent; --i: 1;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fjs.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fjs.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fjs.png&w=256&q=75 2x"
        style="color: transparent; --i: 2;"
        width="120"
      />
      <img
        alt="vs code"
        class="image"
        data-nimg="1"
        decoding="async"
        height="120"
        loading="lazy"
        src="/_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75"
        srcset="/_next/image?url=%2Flogos%2Fadobe.png&w=128&q=75 1x, /_next/image?url=%2Flogos%2Fadobe.png&w=256&q=75 2x"
        style="color: transparent; --i: 3;"
        width="120"
      />
    </div>
  </div>
</body>
`;
